### PR TITLE
PICARD-3000: Children's Music is shown as "Children'S Music" in Picard

### PR DESCRIPTION
**Pull request by zytact**
# Summary

* This is a…
  * [x] Bug fix
  * [ ] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-3000

Genre `Children's Music` is shown as `Children'S Music` using Python's title method.

# Solution

Create a custom function to handle making the genre title case which also handles the apostrophe.

### CODE REVIEW

**Review by phw — CHANGES_REQUESTED** [review]
Thanks for this patch. For this use case it should work. But I wonder if we should use a better function to begin with.

There is an existing title case function as scripting at https://github.com/metabrainz/picard/blob/master/picard/script/functions.py#L964

I think we should factor out the title casing functionality into a function in `picard.util`, e.g. just `def titlecase(s)`. Then this function can be used both here for handling the genres and in the `func_title` scripting function.

This also allows to add tests for the title case function.

**Comment by zytact** [comment]
Thank you for looking into it. I think that would be best way to do it. Should I do all this in one PR or create another PR?

**Comment by phw** [comment]
@zytact Would be allright to do this as part of this PR

**Review by zas — APPROVED** [review]
Nice fix & code improvement, LGTM.

File: picard/track.py [hunk 1]
```diff
@@ -321,6 +324,29 @@ def _customize_metadata(self):
...
+            Examples:
+                >>> titlize("children's music")
+                'Children's Music'
+                >>> titlize("blues")
+                "Blues"
+            """
+            return re.sub(r"[A-Za-z]+('[A-Za-z]+)?",
+                        lambda m: m.group(0)[0].upper() + m.group(0)[1:].lower(),
+                        text)
+
```
Comment (phw) [inline picard/track.py:349]: The `titlize` function here can be removed now that the code form util is being used.

**Review by phw — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution. Looks great. Just the initial titlize function added to track.py can now be removed. Then this is good to merge

File: picard/track.py (hunk 1 above)
Comment (zytact) [reply picard/track.py]: Apologies for the oversight! Thank you for pointing that.

**Review by zas — APPROVED** [review]

**Pull request merged**